src/RawPOSIX/src/safeposix/shm.rs

```diff
@@ -139,9 +143,18 @@ impl ShmMetadata {
             shmtable: interface::RustHashMap::new(),
         }
     }
+    pub fn get_shm_length(&self, shmid: i32) -> Option<usize> {
+        self.shmtable.get(&shmid).map(|segment| segment.get_shm_length())
+    }
+    
 
     pub fn new_keyid(&self) -> i32 {
         self.nextid
             .fetch_add(1, interface::RustAtomicOrdering::Relaxed)
     }
 }
+
```

Comment:
I'm not sure if we need this much abstraction just to get the length. Not sure if theres a better refactor here. Curious to know @qianxichen233 thoughts